onadata/apps/api/tests/viewsets/test_merged_xform_viewset.py

```diff
@@ -540,6 +559,32 @@ def test_get_osm_data_kwargs(self):
             'instance__xform_id': xform.pk
         })
 
+    # pylint: disable=invalid-name
+    def test_merged_with_attachment_endpoint(self):
+        merged_dataset = self._create_merged_dataset()
+        merged_xform = MergedXForm.objects.get(pk=merged_dataset['id'])
+        _make_submissions_merged_datasets(merged_xform)
+
+        # Attachment viewset works ok for filtered datasets
+        attachment_list_view = AttachmentViewSet.as_view({"get": "list"})
+        all_instances = Instance.objects.filter(xform__in=merged_xform.xforms.all())
+        for instance in all_instances:
+            _add_attachments_to_instances(instance)
+        request = self.factory.get(
+            "/?merged_xform=" + str(merged_xform.pk),
+            **self.extra)
+        response = attachment_list_view(request)
+        response_list = json.loads(json.dumps(response.data))
+        serialized_attachments = json.loads(json.dumps(
+            AttachmentSerializer(
```

Comment:
@FrankApiyo Is this not just being repetitive? Is a comparison of `response.data` and `serializer.data` not possible or equivalent?

Reply:
Makes sense, I have refactored; now we compare `response.data` with `serializer.data`